### DISCUSSION TITLE
I am able to insert blocks below my Block Template on the Admin page

### DISCUSSION THREAD

**Discussion by walter74ucla** [WordPress/gutenberg#45500] [General]
### Description

I created a Block Template.  However, I am still able to insert blocks on the admin page below the Block Template.  To me, you use a Block Template to make sure all of the Article Activity happens within the Block Template.

### Step-by-step reproduction instructions

Create a Block Template.
Then try to insert blocks below the Block Template on the New Article Admin Screen.
New blocks are place below the Block Template instead of within the Block Template.

Step 1 - New Article:  add a Title, Subhead, and Paragraph
Step 2 - Add an image below the paragraph
Step 3 - Add another paragraph below the image...This paragraph is inserted below the Block Template (to me, this is a bug)

<img width="1450" alt="Step 1_Screen Shot 2022-10-27 at 12 37 49 PM" src="https://user-images.githubusercontent.com/54418178/198374518-49dc53f1-068d-439d-b6d4-0815ea57d908.png">
<img width="1450" alt="Step 2_Screen Shot 2022-10-27 at 12 41 14 PM" src="https://user-images.githubusercontent.com/54418178/198374551-c59fc5c9-0cae-422e-99dc-9411ee2841d3.png">
<img width="1450" alt="Step 3_Screen Shot 2022-10-27 at 12 44 17 PM" src="https://user-images.githubusercontent.com/54418178/198374588-79dd5dc2-5d72-4a29-a2a3-1d801ebe9891.png">

### Environment info

- WordPress 6.0.3 running [Southern California News Group](http://mercurynews.local/wp-admin/themes.php) theme.
- Gutenberg version 12.0-13.0
- macOS Catalina version 10.15.7

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by talldan** [comment]
> Create a Block Template.

@walter74ucla Would you be able to provide more info on how you created the block template, and the markup of the block template? Thanks.

**Comment by walter74ucla** [comment]
Yes.  Here is the php code:
```php
function register_post_type_post_template() {
	// ... Define $block_template using an array of blocks.  @see how to lock a block above
	$block_template = [
		[ 'mng/print-edit-view-buttons', [] ],
		[ 'mng/mng-post-title', [] ],
		[ 'core/group', [], [
			[ 'mng/subhead', [] ],
			[ 'core/paragraph', [] ],
		] ],
		[ 'mng/warning-block-template', [//This is the "last" unconditional block of the Block Template...This can be removed once we remove the block-list-appender
			'lock' => [
				'move' => true,
				'remove' => true,
			] ]
		],
		//The blocks below are displayed conditionally in this Block Template.
		//The blocks are placed below the unconditional blocks to create the different views.
		[ 'mng/edit-view', [] ],

		//You can also build a Block Template using JavaScript
		//https://wholesomecode.ltd/using-a-template-with-innerblocks-in-the-wordpress-block-editor-gutenberg
	];

	$post_type_object = get_post_type_object( 'post' );
	$post_type_object->template = $block_template;
}
add_action( 'init', 'register_post_type_post_template' );
```

As a work around, I can display: none the block-list-appender class.

**Comment by talldan** [comment]
@walter74ucla By default templates aren't locked and are editable, including being able to insert blocks before, within or after the template.

You can define `$post_type_object->template_lock = 'all'` or `$post_type_object->template_lock = 'insert'` to prevent blocks from being inserted.

**Comment by walter74ucla** [comment]
@talldan Is there a Block Editor component?  There is a Rich Text component:  https://github.com/WordPress/gutenberg/blob/HEAD/packages/block-editor/src/components/rich-text/README.md, but it does not have all of the behavior of the Block Editor like inserting blocks.
This does not work in my application---https://github.com/WordPress/gutenberg/tree/trunk/packages/block-editor

Are there additional working examples that dive deeper into the Block Editor?

**Comment by talldan** [comment]
What is it that you're trying to achieve? Are you looking to implement your own editor?

edit: I'm going to convert this to a github discussion, as issues are for bug/enhancement reports.

**Comment by walter74ucla** [comment]
@talldan Thanks for converting this to a discussion.

Ultimately, I am trying to create a Block Editor Component that can be used inside custom blocks on the post-new.php and post.php admin pages.  In the Classic Editor, we can create Classic Editor meta box via Tiny MCE.  I have attached a picture of the Classic Editor example.

<img width="1450" alt="Screen Shot 2022-11-03 at 5 59 05 AM" src="https://user-images.githubusercontent.com/54418178/199715585-57059769-f806-44d7-acd6-b14ab417ae4b.png">

The reason we do this is to give our editors three views---Web View, Edit View, and Print View via javascript.  We copy data back and forth between the views.
